provider/error.go

```diff
@@ -0,0 +1,25 @@
+package provider
+
+type AlreadyExistsError struct {
+	Msg string
```

Comment:
`Msg` is accessible from another package which is likely not what you want.
So use lower case `msg`

On the other hand you could also just create a simple error variable:

```
var (
   AlreadyExistsError = errors.New("Err: already exits")
   DoesNotExistError = errors.New("Err: does not exist")
)
```

I think this would clean up the code even more.

Reply:
errors.New returns type error, while we relay on new dataTypes for these errors. 
Change Msg to lowercase.